ipfx/dataset/ephys_nwb_data.py

```diff
@@ -61,12 +61,21 @@ def __init__(self,
                  ):
 
         super().__init__(ontology=ontology)
-        load_into_memory = False # not working
-        if load_into_memory:
-            with open(nwb_file, 'rb') as fh:
-                nwb_file = BytesIO(fh.read())
         self.nwb_file = nwb_file
-        self.nwb = NWBHDF5IO(nwb_file, mode='r').read()
+        
+        if isinstance(nwb_file, str):
+            if load_into_memory:
+                with open(nwb_file, 'rb') as fh:
+                    data = BytesIO(fh.read())
+                _h5_file = h5py.File(data, "r")
+                self.nwb = NWBHDF5IO(path=_h5_file.filename, mode="r+",file=_h5_file).read()
+            else:
+                self.nwb = NWBHDF5IO(nwb_file, mode='r').read()
+        elif isinstance(nwb_file, BytesIO):
+            _h5_file = h5py.File(nwb_file, "r")
+            self.nwb = NWBHDF5IO(path=_h5_file.filename, mode="r+",file=_h5_file).read()
+        else:
+            print("Invalid input NWB file, please input either NWB file path or hdf5 obj!")
```

Comment:
maybe 
`raise TypeError(<message>)` 
would be better

Reply:
overall, this entire block of code could be a function that returns nwb object

Reply:
in any case, looks good. I am fine with merging. Could you please merge it tonight?

Reply:
adjust accordingly